### PR TITLE
added support for OAuth bearer access token credential

### PR DESCRIPTION
**Pull request by AnorexicAtticusFinch**

### CODE REVIEW

File: zendesk/zendesk.go [hunk 1]
```diff
@@ -217,7 +217,12 @@ func (z *Client) prepareRequest(ctx context.Context, req *http.Request) *http.Re
 	out := req.WithContext(ctx)
 	z.includeHeaders(out)
 	if z.credential != nil {
-		out.SetBasicAuth(z.credential.Email(), z.credential.Secret())
+		switch z.credential.Type() {
+		case "Basic":
```
Comment (anshal21) [inline zendesk/zendesk.go]: let's make these constants

File: zendesk/credential.go [hunk 2]
```diff
@@ -53,3 +59,35 @@ func (c APITokenCredential) Email() string {
 func (c APITokenCredential) Secret() string {
 	return c.apiToken
 }
+
+// Type is accessor which returns the type
+func (c APITokenCredential) Type() string {
```
Comment (anshal21) [inline zendesk/credential.go:69]: we need to change this to "Basic" as well

File: zendesk/zendesk.go (hunk 1 above)
Comment (AnorexicAtticusFinch) [reply zendesk/zendesk.go]: Made the change 👍

File: zendesk/credential.go (hunk 2 above)
Comment (AnorexicAtticusFinch) [reply zendesk/credential.go:69]: Made the change 👍

**Review by anshal21 — APPROVED** [review]

**Pull request merged**